app/services/sso_service.rb

```diff
@@ -64,7 +64,10 @@ def new_login?
   end
 
   def authn_context
-    REXML::XPath.first(saml_response.decrypted_document, '//saml:AuthnContextClassRef')&.text || 'unknown'
+    if saml_response.decrypted_document
+      authncontext = REXML::XPath.first(saml_response.decrypted_document, '//saml:AuthnContextClassRef')&.text
+    end
+    authncontext || SAML::User::UNKNOWN_AUTHN_CONTEXT
```

Comment:
don't you need to set authncontext variable to nil above the if statement? You might get a nomethoderror, undefined authncontext otherwise, right?

Reply:
d'oh. yeah. All because of a silly line length refactor.